pages/api/products.ts

```diff
@@ -0,0 +1,39 @@
+import nc from "next-connect";
+
+import authMiddleware from "../../src/server/middlewares/auth.middleware";
+import loggerMiddleware from "../../src/server/middlewares/logger.middleware";
+import productService from "../../src/server/services/product.service";
+import validationSchema from "../../src/server/validations/products.validation";
+
+import type { NextApiAuthRequest } from "../../src/server/types/auth.types";
+import type { NextApiResponse } from "next";
+
+const handler = nc<NextApiAuthRequest, NextApiResponse>({})
+  .use(loggerMiddleware)
+  .use(authMiddleware)
+  .get(async (req, res) => {
+    try {
+      const getAllproducts = await productService.getAll(req.user.userId);
+
+      res.status(200).json(getAllproducts);
+    } catch (error) {
+      res.status(500).json(error);
+    }
+  })
+  .post(async (req, res) => {
+    try {
+      const { title, price } =
+        await validationSchema.schemaProductEdit.validate(req.body);
+
+      const { userId } = await validationSchema.schemaId.validate({
+        userId: req.user.userId,
```

Comment:
there is no need to validate userId, since we add it to the request object manually in authMiddleware